application/classes/Ushahidi/Repository/Form/Attribute.php

```diff
@@ -234,13 +233,45 @@ public function getByForm($form_id)
 		return $this->getCollection($results->as_array());
 	}
 
-	/**
+    /**
 	 * @param int $form_id
 	 * @return Entity|FormAttribute
 	 *
-	 * Selects the first attribute of the first stage
+	 * Selects the first attribute of the first stage AFTER $last_attribute_id
+	 * for the form $form_id
 	 * and returns it as a FormAttribute entity
 	 */
+	public function getNextByFormAttribute($form_id, $last_attribute_id)
+	{
+		//FIXME need a better query so it finds the first next attribute in mysql
+		// even when changing stagges... instead of this awful thing
```

Comment:
I think this is a data model issue as much as anything else, attributes should know their form unless we are really ever going to allow them to be used by multiple forms which seems only moderately useful.